addons/sale/models/account_move.py

```diff
@@ -0,0 +1,185 @@
+        analytic_accounts = self.mapped('analytic_account_id')
+
+        # link the analytic account with its open SO by creating a map: {AA.id: sale.order}, if we find some analytic accounts
+        mapping = {}
+        if analytic_accounts:  # first, search for the open sales order
+            sale_orders = self.env['sale.order'].search([('analytic_account_id', 'in', analytic_accounts.ids), ('state', '=', 'sale')], order='create_date DESC')
+            for sale_order in sale_orders:
+                mapping[sale_order.analytic_account_id.id] = sale_order
+
+            analytic_accounts_without_open_order = analytic_accounts.filtered(lambda account: not mapping.get(account.id))
+            if analytic_accounts_without_open_order:  # then, fill the blank with not open sales orders
+                sale_orders = self.env['sale.order'].search([('analytic_account_id', 'in', analytic_accounts_without_open_order.ids)], order='create_date DESC')
+            for sale_order in sale_orders:
+                mapping[sale_order.analytic_account_id.id] = sale_order
+
+        # map of AAL index with the SO on which it needs to be reinvoiced. Maybe be None if no SO found
+        return {move_line.id: mapping.get(move_line.analytic_account_id.id) for move_line in self}
+
+    @api.multi
+    def _sale_prepare_sale_line_values(self, order, price):
+        """ Generate the sale.line creation value from the current move line """
+        self.ensure_one()
+        last_so_line = self.env['sale.order.line'].search([('order_id', '=', order.id)], order='sequence desc', limit=1)
+        last_sequence = last_so_line.sequence + 1 if last_so_line else 100
+
+        fpos = order.fiscal_position_id or order.partner_id.property_account_position_id
+        taxes = fpos.map_tax(self.product_id.taxes_id, self.product_id, order.partner_id)
+
+        return {
+            'order_id': order.id,
+            'name': self.name,
+            'sequence': last_sequence,
+            'price_unit': price,
+            'tax_id': [x.id for x in taxes],
+            'discount': 0.0,
+            'product_id': self.product_id.id,
+            'product_uom': self.product_uom_id.id,
+            'product_uom_qty': 0.0,
+            'is_expense': True,
+        }
+
+    @api.multi
+    def _sale_get_invoice_price(self, order):
+        """ Based on the current move line, compute the price to reinvoice the analytic line that is going to be created (so the
+            price of the sale line).
+        """
+        self.ensure_one()
+
+        unit_amount = self.quantity
+        amount = (self.credit or 0.0) - (self.debit or 0.0)
+
+        if self.product_id.expense_policy == 'sales_price':
+            return self.product_id.with_context(
+                partner=order.partner_id.id,
+                date_order=order.date_order,
+                pricelist=order.pricelist_id.id,
+                uom=self.product_uom_id.id
+            ).price
+        if unit_amount == 0.0:
```

Comment:
Float compare ? (or float is zero that I think exists)

Reply:
Probably, but with which precision ? 
As the product uom is not required on account.move.line, should we use the rounding of the UoM and fallback on `dp.get_precision('Product Unit of Measure')`?